catalog/app/containers/Bucket/Queries/Athena/Athena.tsx

```diff
@@ -341,61 +371,65 @@ function State({ children, queryExecutionId }: StateProps) {
     <WorkgroupsFetcher>
       {({ handleWorkgroupsLoadMore, workgroupsData }) =>
         workgroupsData.case({
-          _: ({ value: workgroups }) => (
-            <QueryResultsFetcher queryExecutionId={queryExecutionId}>
-              {({ queryResultsData, handleQueryResultsLoadMore }) => {
-                const queryExecution = (queryResultsData as requests.AsyncData<
-                  requests.athena.QueryResultsResponse,
-                  requests.athena.QueryExecution | null
-                >).case({
-                  _: () => null,
-                  Ok: ({ queryExecution: qE }) => qE,
-                })
-                const selectedWorkgroup =
-                  workgroup ||
-                  queryExecution?.workgroup ||
-                  workgroups?.defaultWorkgroup ||
-                  ''
-
-                return (
-                  <QueriesFetcher workgroup={selectedWorkgroup} key={queryExecutionId}>
-                    {({
-                      queriesData,
-                      executionsData,
-                      handleQueriesLoadMore,
-                      handleExecutionsLoadMore,
-                    }) => (
-                      <QueryRunner
-                        queryBody={queryRequest || ''}
-                        workgroup={selectedWorkgroup}
-                      >
-                        {({ queryRunData }) =>
-                          children({
-                            customQueryBody,
-                            executionsData,
-                            handleExecutionsLoadMore,
-                            handleQueriesLoadMore,
-                            handleQueryBodyChange: setCustomQueryBody,
-                            handleQueryMetaChange,
-                            handleQueryResultsLoadMore,
-                            handleSubmit,
-                            handleWorkgroupChange,
-                            handleWorkgroupsLoadMore,
-                            queriesData,
-                            queryMeta,
-                            queryResultsData,
-                            queryRunData,
-                            workgroupsData,
-                            workgroup: selectedWorkgroup,
-                          })
-                        }
-                      </QueryRunner>
-                    )}
-                  </QueriesFetcher>
-                )
-              }}
-            </QueryResultsFetcher>
-          ),
+          _: ({ value: workgroups }) =>
+            AsyncResult.Init.is(workgroups) || workgroups?.list?.length ? (
```

Comment:
the first argument passed to this function is `AsyncResult`, so `workgroups` cant be `AsyncResult`, bc its a `value` prop of that `AsyncResult`.
something like this should work:
```
 _: (x) =>
  AsyncResult.Init.is(x) || x.value?.list?.length ? (
```

Reply:
`AsyncResult.Init.is(workgroups)` returns true right now

Reply:
![Screenshot from 2021-05-27 17-17-25](https://user-images.githubusercontent.com/533229/119842612-7e575780-bf0f-11eb-95a5-8e90b8b0ae0b.png)
![Screenshot from 2021-05-27 17-17-36](https://user-images.githubusercontent.com/533229/119842614-7f888480-bf0f-11eb-969a-f0803f499bdc.png)

Reply:
🤔 that's strange, but you can merge if you're confident that it works as expected, tho i'd suggest adding a note to possibly come back to that later

Reply:
probably it's bc when result is pending it has previous state as a value

Reply:
I suppose this is more correct:
```tsx
 _: (x) =>
  AsyncResult.Pending.is(x) || x.value?.list?.length ? (
```

Reply:
then probably `AsyncResult.Init.is(x) || AsyncResult.Pending.is(x) || x.value?.list?.length ? (` (if i understand the intent of this condition correctly)